include/seqan3/io/sam_file/detail/format_sam_base.hpp

```diff
@@ -423,6 +406,20 @@ inline void format_sam_base::read_header(stream_view_type && stream_view,
     auto end = std::ranges::end(stream_view);
     std::vector<char> string_buffer{};
 
+    auto make_tag = [](uint8_t char1, uint8_t char2) constexpr
+    {
+        return static_cast<uint16_t>(char1) | (static_cast<uint16_t>(char2) << CHAR_BIT);
+    };
+
+    auto read_tag = [](auto it, std::array<char, 2> & raw_tag)
+    {
+        raw_tag[0] = *it;
+        ++it;
+        raw_tag[1] = *it;
+        ++it;
+        return it;
+    };
```

Comment:
```suggestion
    auto read_tag = [&it](std::array<char, 2> & raw_tag)
    {
        raw_tag[0] = *it;
        ++it;
        raw_tag[1] = *it;
        ++it;
    };
```

okay let's do enricos suggestion